akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -159,6 +166,14 @@ private[io] object SelectionHandler {
             channelActor ! new ChannelRegistration {
               def enableInterest(ops: Int): Unit = enableInterestOps(key, ops)
               def disableInterest(ops: Int): Unit = disableInterestOps(key, ops)
+              def cancel(): Unit = {
+                key.cancel()
+                // On Windows the selector does not effectively cancel the registration until after the
+                // selector has woken up. Because here the registration is explicitly cancelled, we wake up the
+                // selector to make sure the cancellation (e.g. sending a RST packet for a cancelled TCP connection)
+                // is performed immediately.
+                selector.wakeup()
+              }
```

Comment:
The reason is that `selector` isn't really thread-safe and all kinds of strange things happen when you manipulate things from the outside. Running it in a task on the single pinned dispatcher thread reserved for a selector will make sure all those accesses are serialized.

So, `selector.wakeup` would be fine, but `key.cancel()` probably isn't. On the other hand if you wrap `key.cancel` in the task you don't need the manual wakeup any more as `execute` already takes care of that.